### ISSUE TITLE
Port POSIX `IReactorProcess` implementation to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5968]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5968|
|Type|enhancement|
|Created|2012-09-12 14:45:54Z|

This involves a few different things:

  * The primary implementation module, `twisted.internet.process`
  * At least some parts of `twisted.internet.posixbase` (`spawnProcess` definition) and `twisted.internet.base` (`_checkProcessArgs`)
  * Get `twisted.internet.test.test_process` to pass
  * Get `twisted.test.test_process` to pass
  * Enable process cleanup on Python 3 in `twisted.internet.test.reactormixins`.

<details><summary>Searchable metadata</summary>

```
trac-id__5968 5968
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347461154000000 1347461154000000
changetime__1430296824297252 1430296824297252
version__None None
owner__ 
cc__garrison
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

To make the porting simpler, we might want to do parts of it separately, e.g. only `twisted.internet.process` in [#5987](https://github.com/twisted/twisted/issues/5987).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[#6791](https://github.com/twisted/twisted/issues/6791) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

[#6983](https://github.com/twisted/twisted/issues/6983) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

I believe this is fixed now by [#5987](https://github.com/twisted/twisted/issues/5987).
